File: leetcode/937 Reorder Data in Log Files/solution.java
class Solution {
    public String[] reorderLogFiles(String[] logs) { // 重写数组排序方法！！！
        Arrays.sort(logs, new Comparator<String>(){
            @Override
            public int compare(String o1, String o2) {
                String[] log1 = o1.split(" ", 2); // 以（首个）空格为分隔符，仅分为两部分！！！
                String[] log2 = o2.split(" ", 2);
                boolean isDigit1 = Character.isDigit(log1[1].charAt(0));
                boolean isDigit2 = Character.isDigit(log2[1].charAt(0));
                if (!isDigit1 && !isDigit2) {
                    int cmp = log1[1].compareTo(log2[1]); // 按字典顺序比较两个字符串；小于等于大于分别对应返回-1, 0, 1！！！
                    if (cmp != 0) {
                        return cmp;
                    } else {
                        return log1[0].compareTo(log2[0]);
                    }
                } else {
                    return isDigit1 ? (isDigit2 ? 0 : 1) : -1; // 在Java重写排序中，返回-1表示不交换顺序，0表示相同时不交换，1表示交换！！！
                }
            }
        });
        return logs;
    }
}

File: leetcode/937 Reorder Data in Log Files/solution.go
func reorderLogFiles(logs []string) []string { // 重写数组排序方法！！！
	sort.SliceStable(logs, func(i, j int) bool { // import sort！！！
		log1 := strings.SplitN(logs[i], " ", 2) // import strings; 以（首个）空格为分隔符，仅分为两部分！！！
		log2 := strings.SplitN(logs[j], " ", 2)
		isDigit1 := unicode.IsDigit(rune(log1[1][0])) // import unicode; 注意输入类型需要强转为rune！！！
		isDigit2 := unicode.IsDigit(rune(log2[1][0]))
		if !isDigit1 && !isDigit2 {
			if log1[1] != log2[1] {
                return log1[1] < log2[1] // 按字典顺序比较两个字符串；小于则返回true！！！
            } else {
                return log1[0] < log2[0]
            }
		} else {
            return !isDigit1 && isDigit2  // 在Go重写排序中，返回true表示该排序规则满足需要（不交换），返回false表示该排序规则不满足需要（交换）！！！
        }
	})
	return logs
}